### PR TITLE
PRL-3560:Added changes to add party id

### PR DESCRIPTION
**Pull request by khurshedk**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [x] the `keep-helm` label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/models/complextypes/manageorders/ServedParties.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder(toBuilder = true)
+@AllArgsConstructor
+public class ServedParties {
+
+    @JsonProperty("partyId")
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/models/complextypes/manageorders/ServedParties.java:13]: Do we need this @JsonProperty?

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java [hunk 2]
```diff
@@ -128,6 +130,7 @@ public class ManageOrderService {
     public static final String RECIPIENTS_OPTIONS = "recipientsOptions";
 
     public static final String OTHER_PARTIES = "otherParties";
+    public static final String SERVEDPARTIES = "servedParties";
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java:133]: SERVED_PARTIES

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java [hunk 3]
```diff
@@ -1344,6 +1350,38 @@ private void servedC100Order(CaseData caseData, List<Element<OrderDetails>> orde
...
+            .getRecipientsOptionsOnlyC47a() != null) {
+            servedParties = dynamicMultiSelectListService
+                .getServedPartyDetailsFromDynamicSelectList(caseData.getManageOrders()
+                                                         .getRecipientsOptionsOnlyC47a());
+        }
+        if (caseData.getManageOrders().getChildOption() != null) {
+            servedParties.addAll(dynamicMultiSelectListService
+                                     .getServedPartyDetailsFromDynamicSelectList(caseData.getManageOrders()
+                                                                                     .getChildOption()));
+        }
```
Comment (AlokDatta) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java:1371]: For FL401, do we bother about childs involved while drawing order? If not, we can put another & for case type please.

**Comment by AlokDatta** [comment]
there is 1 code smell https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&sinceLeakPeriod=true&pullRequest=1397&id=prl-cos-api

File: src/main/java/uk/gov/hmcts/reform/prl/models/complextypes/manageorders/ServedParties.java (hunk 1 above)
Comment (khurshedk) [reply src/main/java/uk/gov/hmcts/reform/prl/models/complextypes/manageorders/ServedParties.java]: removed the Json property element.

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java (hunk 2 above)
Comment (khurshedk) [reply src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java]: renamed to SERVED_PARTIES

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java [hunk 4]
```diff
@@ -1344,6 +1351,39 @@ private void servedC100Order(CaseData caseData, List<Element<OrderDetails>> orde
...
+                                                         .getRecipientsOptionsOnlyC47a());
+        }
+        if (C100_CASE_TYPE.equalsIgnoreCase(CaseUtils.getCaseTypeOfApplication(caseData))
+            && caseData.getManageOrders().getChildOption() != null) {
+            servedParties.addAll(dynamicMultiSelectListService
+                                     .getServedPartyDetailsFromDynamicSelectList(caseData.getManageOrders()
+                                                                                     .getChildOption()));
+        }
+
+        if (caseData.getManageOrders().getOtherParties() != null) {
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java:1375]: Other party applies only for C100 case, should we add check here?

**Review by shashisk97 — APPROVED** [review]
Check my comment nd approved

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java (hunk 4 above)
Comment (khurshedk) [reply src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java]: @shashisk97 Added condition to add other parties only in case of C100

**Review by AlokDatta — APPROVED** [review]

**Pull request merged**